### DISCUSSION TITLE
Supporting reusable fragments in TOML

### DISCUSSION THREAD

**Discussion by yairlenga** [toml-lang/toml#1012] [Ideas]
In the JSON land, it's common to have URL values, which sometimes are resolved to references into JSON objects. Common example is the "$ref": "URL", which will use the '#fragment' with JSON pointer format: "http://site.com/foo/bar.json#my/object".

Is there any standard/proposal to have this as built-in type, distinct from the string. In particular, If there is standard way to refer to other part of the document, it will be possible to create "reusable" parts, similar in spirit to what's being used in JSON schema:

Which will result in tweety.type.legs == 2, and sooby_doo.type.tail == true

```
[type.bird]
    wings: true
    legs: 2

[type.dog]
    legs: 4
    tail: true

[tweety]
    type: /type/bird

[scooby_doo]
    type: /type/dog

```

**Comment by eksortso** [comment]
To date, no such reference format exists _within_ the TOML standard. If it did exist, it would exist _outside_ of the standard, and could be an application of the standard.

We want to keep TOML as simple as possible. If you want to implement a pointer-style system, you could express a pointer, by your own standards, within a TOML string; and then later, you can interpret those pointer strings in your TOML document to navigate your own table hierarchy.

That's basically how JSON Pointer uses JSON anyway. See [RFC 6901](https://www.rfc-editor.org/rfc/rfc6901) for details. Section 5 shows how pointers can be defined within strings in JSON. That doesn't make pointers part of JSON. It just shows that pointers can be expressed in JSON and then subsequently be treated as references by other tools, because your application expects to find and use pointer strings in that way.

**Reply by yairlenga** [reply]
That’s very reasonable approach, at least until there is clear, strong demand for this functionality.
I can only wish that there was a way to express “typed” string to distinguish a “regular” string, vs a reference (or any other type) string - something like:

type: pointer’/type/bird’
VS
Type: ‘pointer:/type/bird’

But this could be a subject for bother ticket. I will close this ticket.
